oidc.go

```diff
@@ -187,6 +199,46 @@ func (h *Headscale) OIDCCallback(ctx *gin.Context) {
 		return
 	}
 
+	// If AllowedDomains is provided, check that the authenticated principal ends with @<alloweddomain>.
+	if len(h.cfg.OIDC.AllowedDomains) > 0 {
+		domainFound := false
+
+		for _, domain := range h.cfg.OIDC.AllowedDomains {
+			if strings.HasSuffix(claims.Email, fmt.Sprintf("@%s", domain)) {
+				domainFound = true
+
+				break
+			}
+		}
```

Comment:
I feel like this can be rewritten to something more generic that strips the claim to the hostname and use a generic function for String in list? (This would also apply to the user bellow).

I am certain we have a couple of those spread around in the code, at least one.